### DISCUSSION TITLE
why do we need adj.t()?

### DISCUSSION THREAD

**Discussion by liangbright** [pyg-team/pytorch_geometric#4584] [Q&A]
I am curious to know why we need to input "adj.t()" to a layer:
"https://pytorch-geometric.readthedocs.io/en/latest/notes/sparse_tensor.html
out2 = conv(x, adj.t())"

It seems to be easy for the user to input adj, and the conv layer can transpose it if necessary

**Comment by Padarn** [comment]
I had a similar confusion, but the transposed matrix is the more useful one to store. You can take a look at an earlier discussion where I discussed his with @rusty1s https://github.com/pyg-team/pytorch_geometric/discussions/4248#discussioncomment-2349167.

As for he API taking `adj`.. I guess I'm not qualified to say, but it seems better to generally encourage the user to keep `adj_t` around rather than `adj` given the above reasoning.
